Rest/Controller/ContentTypeController.php

```diff
@@ -84,35 +82,19 @@ protected function prepareContentByContentTypeIds($contentTypeIds, $responseType
                 $criteria[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
             }
 
-            $siteAccess = $request->get('sa', $this->siteAccess->name);
-            $rootLocationId = $this->configResolver->getParameter('content.tree_root.location_id', null, $siteAccess);
-            $criteria[] = new Criterion\Subtree($this->locationService->loadLocation($rootLocationId)->pathString);
+            $criteria[] = new Criterion\Subtree($rootLocationPathString);
 
             $query = new Query();
             $query->query = new Criterion\LogicalAnd($criteria);
+            $query->limit = $pageSize;
+            $query->offset = $offset;
```

Comment:
Is it possible to use Pagerfanta here as in case of Admin UI? Or is Pagerfanta dedicated to Twig? Don't know :) Just random thought, so disregard if it doesn't make sense.

Reply:
Yes, but as another PR